### PR TITLE
feat: release 2.6

### PR DESCRIPTION
**Pull request by spacewander**
Signed-off-by: spacewander <spacewanderlzx@gmail.com>

### Pre-submission checklist:

* [ ] Did you explain what problem does this PR solve? Or what new features have been added?
* [ ] Have you added corresponding test cases?
* [ ] Have you modified the corresponding document?
* [ ] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix/tree/master#community) first**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -41,6 +42,41 @@ title: Changelog
...
+- :sunrise: feat: support client certificate verification [#4034](https://github.com/apache/apisix/pull/4034)
+- :sunrise: feat: add nacos support [#3820](https://github.com/apache/apisix/pull/3820)
+- :sunrise: feat: patch tcp.sock.connect to use our DNS resolver [#4114](https://github.com/apache/apisix/pull/4114)
+
+### Plugin
+
+- :sunrise: feat(redirect): support uri encoding [#4244](https://github.com/apache/apisix/pull/4244)
+- :sunrise: feat(key-auth): allow customizing header [#4013](https://github.com/apache/apisix/pull/4013)
+- :sunrise: feat(response-rewrite): allow using variable in the header [#4194](https://github.com/apache/apisix/pull/4194)
+- :sunrise: feat(ext-plugin): finish the first version [#4183](https://github.com/apache/apisix/pull/4183)
```
Comment (membphis) [inline CHANGELOG.md:68]: add more comments: 

`With this feature, APISIX can support Java, Golang and other languages to implement custom plug-ins.`

**Review by membphis — APPROVED** [review]

**Comment by dmsolr** [comment]
I think `docs:` is a little better than `feat:`. :)

**Comment by spacewander** [comment]
It is not a doc-only change.

**Review by tokers — APPROVED** [review]

**Review by Firstsawyou — APPROVED** [review]

**Review by juzhiyuan — APPROVED** [review]
LGTM

**Pull request merged**